lib/pf/pfqueue/stats.pm

```diff
@@ -0,0 +1,128 @@
+package pf::pfqueue::stats;
+
+=head1 NAME
+
+pf::pfqueue::stats - Object for collecting pfqueue stats
+
+=head1 DESCRIPTION
+
+
+
+=cut
+
+use Moo;
+use namespace::autoclean;
+use pf::config::pfqueue;
+use pf::constants::pfqueue qw($PFQUEUE_COUNTER $PFQUEUE_QUEUE_PREFIX $PFQUEUE_EXPIRED_COUNTER);
+use Redis::Fast;
+
+has redis => (is => 'ro', builder => 1, lazy => 1);
+
+=head1 METHODS
+
+=head2 my $counters = $self->counters()
+
+Get all the task counters
+
+=cut
+
+sub counters {
+    my ($self) = @_;
+    return $self->_get_counters_for($PFQUEUE_COUNTER);
+}
+
+=head2 my $miss_counters = $self->miss_counters()
+
+Get all the miss counters
+
+=cut
+
+sub miss_counters {
+    my ($self) = @_;
+    return $self->_get_counters_for($PFQUEUE_EXPIRED_COUNTER);
+}
+
+=head2 $counters = $self->_get_counters_for($counter_name)
+
+Get all the counters for a namespace
+
+=cut
+
+sub _get_counters_for {
+    my ($self, $counter_name) = @_;
+    my $redis = $self->redis;
+    my %counters = $redis->hgetall($counter_name);
+    my @counters = map { &_counter_map(\%counters, $_) } sort keys %counters;
+    return \@counters;
+}
+
+=head2 my $counter = _counter_map($counters, $key)
+
+Utility function for normalizing the counter data
+
+=cut
+
+sub _counter_map {
+    my ($counters, $key) = @_;
+    $key =~ /^\Q$PFQUEUE_QUEUE_PREFIX\E([^:]+):(.*)$/;
+    my $queue = $1;
+    my $name = $2;
+    my %counter = (
+        name => $name,
+        queue => $queue,
+        count => $counters->{$key},
+    );
+    return \%counter;
+}
+
+
+=head2 _build_redis
+
+=cut
+
+sub _build_redis {
+    my ($self) = @_;
+    return Redis::Fast->new( %{$ConfigPfQueue{consumer}{redis_args}});
```

Comment:
There are multiple place where this is done, this should be in a central point so its not duplicated